File: eero-pand/koerad/Dog.php
<?php
declare(strict_types=1);
ini_set ( 'display_errors', '1' );

if (defined ( 'E_DEPRECATED' )) {
	error_reporting ( E_ALL & ~ E_DEPRECATED & ~ E_STRICT );
} else {
	error_reporting ( E_ALL & ~ E_STRICT );
}
class Dog {
      public static function askName() {
      	     echo "Auh!\n";
	     $name = readline("Mis on sinu nimi? ");
	     echo sprintf(
	     	  'Kas su nimi on tõesti %1$s?%2$s',
		  $name, // 1
		  "\n" // 2
		  );
		  if (strlen($name) > 10) {
		     echo "See on pikk nimi.\n";
		  } else {
		     echo "See on lühike nimi.\n";
}
		$lengthOfName = strlen($name);
		if ($lengthOfName > 10) {
		     echo "See on pikk nimi.\n";		
		} else if ($lengthOfName > 5) {
		     echo "See on keskmise pikkusega nimi.\n";
		} else {
		     echo "See on lühike nimi.\n";
		}
		switch ($lengthOfName) {
			case 5: {
			     echo "Sama palju korruseid on meie majal.\n";
			     break;
			}
			case 2: {
			     echo "Nii palju erinevaid numbreid tunneb arvuti.\n";
			     break;
			}
			default: {
			     echo "Öösiti on külmem kui väljas.\n";
			}
		}

		echo sprintf(
		     'Sinu nimes sisaldub t-täht %1$u korral. %2$s',
		     Dog::countCharacter($name, 't'), // 1
		     "\n" // 2
		);
		echo sprintf(
		     'Sinu nimes sisaldub t-täht %1$u korral. %2$s',
		     Dog::countCharacterWithWhile($name, 't'), // 1
		     "\n" // 2
		);
	 }
		  public static function countCharacter(string $string, string $character): int {
		  	 $numberOfOccurencies = 0;
			 for ($numberOfCharacter = 0; $numberOfCharacter < strlen($string); $numberOfCharacter++) {
			     if (substr($string, $numberOfCharacter, 1) === $character) {
			     	$numberOfOccurencies++;
				}
				}
				return $numberOfOccurencies;
			}
			public static function countCharacterWithWhile(string $string, string $character): int {
			       $numberOfCharacter = 0;
			       $lengthOfString = strlen($string);
			       $numberOfOccurencies = 0;
			       while ($numberOfCharacter < $lengthOfString) {
			       	     if (substr($string, $numberOfCharacter, 1) === $character) {
				     	$numberOfOccurencies++;
				}
				$numberOfCharacter++;
				}
				return $numberOfOccurencies;
			}
}
Dog::askName();

File: eero-pand/koerad/Dog.java
import java.util.Scanner;
class Dog {
    public static String askName() {
	System.out.println("Mis on sinu nimi?");
	Scanner scanner = new Scanner(System.in);
	String name = scanner.nextLine();
	scanner.close();
	System.out.println("Kas su nimi on tõesti " + name + "?");
	int lengthOfName = name.length();
	if (lengthOfName > 10) {
	    System.out.println("Küll sul on pikk nimi!");
	} else if (lengthOfName > 5) {
	    System.out.println("Küll sul on mõõdukas nimi!");
	} else {
	    System.out.println("Küll sul on lühike nimi!");
	}
	switch (lengthOfName) {
	case 7: {
	    System.out.println("Nii palju on pöialpoisse!");
	    break;
	}
	case 2: {
	    System.out.println("Nii palju eri numbreid tunneb arvuti.");
	    break;
	}
	default: {
	    System.out.println("Ma ei viitsi isegi mitte võrrelda!");
	}
	}
	return name;
    }
    public static void main(String[] args) {
	String name = Dog.askName();
	System.out.println("Sinu nimes sisaldub a-täht " + String.valueOf(Dog.countCharacter(name, 'a')) + " korral.");
	System.out.println("Kontroll: " + Dog.countCharacterWithWhile(name, 'a'));
    }

    public static int countCharacter(String string, char character) {
	int numberOfOccurencies = 0;
	for (int numberOfCharacter = 0; numberOfCharacter < string.length(); numberOfCharacter++) {
		if (string.charAt(numberOfCharacter) == character) {
			numberOfOccurencies++;
		}
	}
	return numberOfOccurencies;
    }

    public static int countCharacterWithWhile(String string, char character) {
	int numberOfOccurencies = 0;
	int numberOfCharacter = 0;
	int lengthOfString = string.length();
	while (numberOfCharacter < lengthOfString) {
	    if (string.charAt(numberOfCharacter) == character) {
		numberOfOccurencies++;
	    }
	    numberOfCharacter++;
	}
	return numberOfOccurencies;
    }
    
}
